### ISSUE TITLE
[0.9.2] Balanced double quotes allow for unescaped spaces

### ISSUE THREAD

**Issue by mjdesa** [influxdb/influxdb#3465]
If a field key has balanced quotes 
```
insert fail value=1,"test this"=1
```
it successfully writes a field key of `"test this"` without anything being escaped, however if the quotes are unbalanced or non-existent they must be escaped

```
> insert fail value=1,"test\ =1
ERR: unable to parse ' fail value=1,"test\ =1': unbalanced quotes

> insert fail value=1,\"test\ =1
```

**Comment by mjdesa** [comment]
It works as expected or single quotes

```
> insert fail value=1,'test=1
```

is successful.

**Comment by lokeshverizon** [comment]
the issue is, whenever a message has new line, it gives this error. Scanner should be smart to handle this case.  

unable to parse 'abc_logs,appname=efg msg="node-38eaa7157a74|10.1.20.1|TORRANCE|1|INFO|13|abc|efg|utag|consumer.go:150|2015/07/31 18:03:48.993878379|Commited offset 7499': unbalanced quotes

**Comment by beckettsean** [comment]
@lokeshverizon line breaks have syntactic meaning in the line protocol. Can you please copy/paste your exact write command? 

If your write command is all on one line with no newline characters, then the issue is just a duplicate of what Michael posted in the first comment. There is an un-matched double-quote in the write statement.

**Comment by lokeshverizon** [comment]
I am using go client API for writing batchPoints. 

                bps := influx.BatchPoints{
                        Points:          pts,
                        Database:        DATABASE_NAME,
                        RetentionPolicy: RETENTION_POLICY,
                }

Here in my batchPoint I am creating a field which has a newline. Shouldn't this API take care of this?

**Comment by beckettsean** [comment]
@lokeshverizon yes, it should take care of properly quoting/escaping the newline in a field value. That's not what this issue #3465 is about, though Please open a new issue for the failure of the Go client to handle the newline character in a field value.

**Comment by jsternberg** [comment]
This issue looks like it is fixed by #5158.

That PR treats quotes in field keys as just a normal character. That means that previously this was accepted:

```
fail value=1,"test this"=1
```

Now it won't be accepted because quotes aren't special in the field key. But this now gets parsed perfectly fine:

```
fail value=1,"test\ this=1
```

I'm closing this ticket since the original issue is resolved by that PR.

@lokeshverizon it looks like your issue is a separate one. Can you make an issue so that one can be tracked separately?
